### DISCUSSION TITLE
Add if/then to Smart Key functionality

### DISCUSSION THREAD

**Discussion by joshthree** [BartoszCichecki/LenovoLegionToolkit#1423] [Ideas]
I currently use my smart key to quickly switch between high performance mode for and low performance mode for battery.  I like that I can use it to change a bunch settings at once.  But it is mildly annoying that I have to use both the single tap and double tap to have both functions.

I think it would be great if I could have an if/then statement.  Say, If (custom preset is high performance) then (do a bunch of things to switch to low power mode, including changing the custom preset to low power mode) else (do a bunch of things to switch to high power mode).  Then I could have that be single tap and have space for another command.

**Comment by BartoszCichecki** [comment]
If/then is a neat idea, but it would require quite a bit of work into Actions. It might be implemented, depending on the demand, but because of the amount of work, I cannot say when.

> between high performance mode for and low performance mode for battery

What do you mean by that?

**Reply by joshthree** [reply]
High performance is high power to CPU and GPU, high refresh rate, and other settings that improve performance for high power things.  Low performance is low power to CPU and GPU, low refresh rate, and other similar settings to maximize battery life for light work away from the charger.  I created a custom profile for high and low performance.

**Reply by joshthree** [reply]
I do hope you choose to implement if/then.  I think it could be if (condition) then (action) else (other action).  It doesn't need to allow you to do a series of actions because you can call a macro.

**Reply by BartoszCichecki** [reply]
Why not just configure Actions that would do it automatically when you plug in/out AC adapter?

**Reply by joshthree** [reply]
I prefer manual controls for things like this.  The main problem with it happening automatically is that it takes time to change the refresh rate and I do occasionally accidentally unplug my computer while gaming.  Plus I don't always use that setting on battery.  I only have that setting for when I expect to be away from a charger for an extended period of time.
